src/Block/Service/AbstractBlockService.php

```diff
@@ -31,21 +32,29 @@
     protected $name;
 
     /**
-     * @var EngineInterface
+     * @var Environment
      */
     protected $templating;
 
     /**
-     * @param string          $name
-     * @param EngineInterface $templating
+     * @param string      $name
+     * @param Environment $templating
+     *
+     * @throws \TypeError when 2nd argument is not Twig\Environment
      */
-    public function __construct($name = null, EngineInterface $templating = null)
+    public function __construct($name, $templating)
     {
-        if (null === $name || null === $templating) {
+        if ($templating instanceof EngineInterface) {
```

Comment:
Do we need to add deprecation messages on master? can't we just move to twig direclty? wdyt @sonata-project/contributors @covex-nn ?

Reply:
I think that is a better option.

Reply:
At the begining i thought, that if i will remove type hint from `__construct` of AbstractBlockService, it will be easier to updgrade all other bundles, because they will be compatible with 3.x and with dev-master of BlockBundle.

I searched in `src/Block` directory inside each bundle and found, that there are 12 bundles, containing `Blocks`:
1. 7 bundles contain at least ont `Block` with extended `__construct` (with more than 2 parameters): AdminBundle, DoctrineORMAdminBundle, MediaBundle, NewsBundle, PageBundle, SeoBundle, TimelineBundle.
2. 4 bundles have `Blocks` without `__construct`: CommentBundle: DashboardBundle, FormatterBundle, TranslationBundle
3. DoctrinePhpcrAdminBundle contain `Block`, that extends deprecated BaseBlockService class

So, first 7 bundles will not be compatible with new AbstractBlockService::__construct =(( May be it is too much and a should change `EngineInterface $templating` to `Environment $templating`?

Reply:
Well, 4.x does not have to be BC with 3.x? /cc @sonata-project/contributors

Reply:
If I understand correctly, the issue is that some bundle need to be compatible with 2 versions of some other bundles, right?